### DISCUSSION TITLE
Problem with bandcamp iframe embeds in next.

### DISCUSSION THREAD

**Discussion by marcusprice** [vercel/next.js#17093] [Help]
Hi there! This is a bit of a weird one for me, so I thought I would reach out to the community for help. I am building a music review blog for someone and decided to build it in next.js. When I add an iframe to embed an album from bandcamp (iframe tag is generated by bandcamp) there is some unexpected behavior. Most of the time when the page renders the album art is transparent (unwanted) and will not appear until the user clicks play on the album. Sometimes it will render correctly and the album art is displayed.

I examined the html of the embed in the iframe with chrome dev tools and can see what's happening: an inline style for a background image (the album art) is supposed to be applied to the album art container div. I can see that on the unsuccessful renders where it's transparent there are no inline styles on the art container until I click play on the album - then they are dynamically added. I can also see on successful renders that the inline styles are there without having to click play on the album. If I embed multiple albums into a page some albums will render successfully with art and others won't.

I realize this sounds like/is most likely a bug with bandcamp and their embeds (which is why I didn't submit an issue), but the weird thing is that as far as I can tell this _only_ happens with next.js. This issue doesn't happen with static html files or even CRA. I also verified that this happens even with a blank next.js project with only the bandcamp iframe inserted into a div with dangerouslySetInnerHTML, so I don't believe it's because of any code I've written.

Does anyone have any ideas of why this is happening?

**Comment by jnv** [comment]
Hi, looks like you spent a good time investigating the issue. Would you mind sharing a minimal example to show the issue so we can dig deeper into it?

From the description of the issue, what separates Next.js from CRA and plain static page is React rehydration. It sometimes causes some headaches, but personally I haven't encountered an issue with embedding iframes per se.

**Reply by marcusprice** [reply]
Thank you for your response. So in making an example for you, I realized that this behavior isn't being replicated in the production build - only in the development version. This is weird because I swear I got this same behavior with a production build as well when I tried it this last weekend, but perhaps I'm mistaken.

If you are curious here is the repo for the example I made: https://github.com/marcusprice/bandcamp-embed-example
And a live example where it appears to be working: https://bandcamp-embed-example.vercel.app/

But like I said everything seems to work as expected in the production build so I guess I won't worry about it. Thanks again for your response and cool to learn about rehydration!

**Reply by marcusprice** [reply]
Actually I take that back, it appears it is actually happening on the production build now that I've viewed it enough times just not as much.

**Comment by SamuelBanya** [comment]
@marcusprice 
Hey Marcus,

I am doing a very similar thing for my own website. 

How did you handle working around CORS issues while working with Bandcamp iFrames in NextJS?

Asking since it doesn't seem like you have your repo publicly available anymore since your link above is dead ^

Thanks.
